### PR TITLE
Always enable experimental features

### PR DESCRIPTION
**Pull request by thaJeztah**
replaces / closes https://github.com/docker/cli/pull/1355

The CLI disabled experimental features by default, requiring users to set a configuration option to enable them.

Disabling experimental features was a request from Enterprise users that did not want experimental features to be accessible.

We are changing this policy, and now enable experimental features by default. Experimental features may still change and/or removed, and will be highlighted in the documentation and "usage" output.

For example, the `docker manifest inspect --help` output now shows:

    EXPERIMENTAL:
      docker manifest inspect is an experimental feature.

      Experimental features provide early access to product functionality. These features
      may change between releases without warning or can be removed entirely from a future
      release. Learn more about experimental features: https://docs.docker.com/go/experimental/
**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: cli-plugins/manager/metadata.go [hunk 1]
```diff
@@ -24,5 +24,6 @@ type Metadata struct {
 	URL string `json:",omitempty"`
 	// Experimental specifies whether the plugin is experimental.
 	// Experimental plugins are not displayed on non-experimental CLIs.
+	// TODO decide if we want to annotate plugins that are experimental
```
Comment (thaJeztah) [inline cli-plugins/manager/metadata.go:27]: /cc @chris-crone @justincormack

File: kubernetes/check.go [hunk 2]
```diff
@@ -24,6 +24,7 @@ const (
 )
 
 // GetStackAPIVersion returns the most appropriate stack API version installed.
+// TODO remove "experimental" argument now that experimental-cli features are enabled by default
```
Comment (thaJeztah) [inline kubernetes/check.go:27]: /cc @chris-crone

File: kubernetes/check.go [hunk 3]
```diff
@@ -33,6 +34,7 @@ func GetStackAPIVersion(serverGroups discovery.ServerGroupsInterface, experiment
 	return getAPIVersion(groups, experimental)
 }
 
+// TODO remove "experimental" argument now that experimental-cli features are enabled by default
```
Comment (thaJeztah) [inline kubernetes/check.go:37]: /cc @chris-crone (let me know if I should just remove it here)

File: cli-plugins/manager/metadata.go (hunk 1 above)
Comment (justincormack) [reply cli-plugins/manager/metadata.go:27]: I dont think we want to annotate them.

**Review by justincormack — APPROVED** [review]

**Comment by thaJeztah** [comment]
Looks like I missed some tests; updating 

```
You can run 'go test . -test.update-golden' to automatically update testdata/docker-help-helloworld.golden to the new expected value.'
``

This test looks flaky (saw it failing on a few runs);

```
=== Failed
=== FAIL: cli/command/container TestRemoveForce/without_force (0.00s)
    --- FAIL: TestRemoveForce/without_force (0.00s)
        rm_test.go:37: assertion failed: 
            --- removed
            +++ →
            {[]string}[?->0]:
            	-: <non-existent>
            	+: "mycontainer"
            

=== FAIL: cli/command/container TestRemoveForce (0.00s)
```

File: kubernetes/check.go (hunk 2 above)
Comment (thaJeztah) [reply kubernetes/check.go]: removed

File: kubernetes/check.go (hunk 3 above)
Comment (thaJeztah) [reply kubernetes/check.go]: removed

File: cli-plugins/manager/metadata.go (hunk 1 above)
Comment (thaJeztah) [reply cli-plugins/manager/metadata.go]: Removed the TODO, and marked the field deprecated

**Comment by thaJeztah** [comment]
FWIW, I preserved the `Experimental:      true` in the output of `docker version`. It's likely that there are scripts that use this output to check if the  cli is configured with experimental features, and we don't want to break those (we could mark that output as deprecated as well, and remove it at some point though)

**Review by silvin-lubecki — APPROVED** [review]
LGTM

**Pull request merged**